lib/db_badger.go

```diff
@@ -0,0 +1,213 @@
+package lib
```

Comment:
This file has no dependencies on the lib directory! You can move the files in this PR to their own packages under the root directory of the project. I suggest a directory structure like this: `core/storage/badger`.

This way, the code in this PR can decoupled from the lib package, and imported appropriately where needed.